diff --git a/CMakeLists.txt b/CMakeLists.txt
index b412dc7..6e5a04a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -90,6 +90,8 @@ include_directories(${CMAKE_CURRENT_BINARY_DIR} ${CMAKE_SOURCE_DIR})
 set(ZLIB_PUBLIC_HDRS
     ${CMAKE_CURRENT_BINARY_DIR}/zconf.h
     zlib.h
+    contrib/minizip/unzip.h
+    contrib/minizip/ioapi.h
 )
 set(ZLIB_PRIVATE_HDRS
     crc32.h
@@ -118,6 +120,8 @@ set(ZLIB_SRCS
     trees.c
     uncompr.c
     zutil.c
+    contrib/minizip/unzip.c
+    contrib/minizip/ioapi.c
 )
 
 if(NOT MINGW)
diff --git a/contrib/minizip/ioapi.c b/contrib/minizip/ioapi.c
index 814a6fd..0ca29db 100644
--- a/contrib/minizip/ioapi.c
+++ b/contrib/minizip/ioapi.c
@@ -231,8 +231,7 @@ static int ZCALLBACK ferror_file_func (voidpf opaque, voidpf stream)
     return ret;
 }
 
-void fill_fopen_filefunc (pzlib_filefunc_def)
-  zlib_filefunc_def* pzlib_filefunc_def;
+void fill_fopen_filefunc (zlib_filefunc_def* pzlib_filefunc_def)
 {
     pzlib_filefunc_def->zopen_file = fopen_file_func;
     pzlib_filefunc_def->zread_file = fread_file_func;
diff --git a/contrib/minizip/unzip.c b/contrib/minizip/unzip.c
index 3036b47..61d8c6b 100644
--- a/contrib/minizip/unzip.c
+++ b/contrib/minizip/unzip.c
@@ -69,7 +69,7 @@
 #include <string.h>
 
 #ifndef NOUNCRYPT
-        #define NOUNCRYPT
+        // #define NOUNCRYPT
 #endif
 
 #include "zlib.h"
diff --git a/zlib.map b/zlib.map
index 31544f2..df26ca6 100644
--- a/zlib.map
+++ b/zlib.map
@@ -15,7 +15,6 @@ ZLIB_1.2.0 {
     zcfree;
     z_errmsg;
     gz_error;
-    gz_intmax;
     _*;
 };
 
